src/components/profile/share/share-profile-dialog.tsx

```diff
@@ -45,53 +47,86 @@ export const ShareProfileDialog = ({
-            handleUrlToClipboard={handleUrlToClipboard}
-          />
-        )}
-        {activeView === 'Instagram' && (
-          <Instagram
-            username={username}
-            setActive={setActive}
-            handleUrlToClipboard={handleUrlToClipboard}
-          />
-        )}
-        {activeView === 'X' && (
-          <XTwitter
-            username={username}
-            setActive={setActive}
-            handleUrlToClipboard={handleUrlToClipboard}
-          />
-        )}
+        <motion.div
+          // TODO: address this hacky solution for the first view. For some reason, the height is not being calculated correctly when the dialog is opened.
+          animate={{
+            height: activeView === 'ShareYourProfile' ? '372px' : contentHeight,
+          }}
+          className='overflow-hidden'
+        >
+          <div ref={contentRef}>
+            {activeView === 'ShareYourProfile' && (
+              <AnimatedContainer>
+                <ShareYourProfile
+                  username={username}
+                  setActive={setActive}
+                  handleUrlToClipboard={handleUrlToClipboard}
+                />
+              </AnimatedContainer>
+            )}
+            {activeView === 'AddToSocials' && (
+              <AnimatedContainer>
+                <AddToSocials setActive={setActive} />
+              </AnimatedContainer>
+            )}
+            {activeView === 'ShareMyProfileTo' && (
+              <AnimatedContainer>
+                <ShareMyProfileTo
+                  username={username}
+                  setActive={setActive}
+                  handleUrlToClipboard={handleUrlToClipboard}
+                />
+              </AnimatedContainer>
+            )}
+            {activeView === 'ShareOnSocials' && (
+              <AnimatedContainer>
+                <ShareOnSocials
+                  username={username}
+                  setActive={setActive}
+                  handleUrlToClipboard={handleUrlToClipboard}
+                />
+              </AnimatedContainer>
+            )}
+            {activeView === 'Instagram' && (
+              <AnimatedContainer>
+                <Instagram
+                  username={username}
+                  setActive={setActive}
+                  handleUrlToClipboard={handleUrlToClipboard}
+                />
+              </AnimatedContainer>
+            )}
+            {activeView === 'X' && (
+              <AnimatedContainer>
+                <XTwitter
+                  username={username}
+                  setActive={setActive}
+                  handleUrlToClipboard={handleUrlToClipboard}
+                />
+              </AnimatedContainer>
+            )}
+          </div>
+        </motion.div>
       </DialogContent>
     </Dialog>
   );
 };
+
+/** All this does is animate the opacity of each component and adds a delay for better timing */
+const AnimatedContainer = ({ children }: { children: React.ReactNode }) => {
```

Comment:
```suggestion
const FadeIn = ({ children }: { children: React.ReactNode }) => {
```

imo, "Container" is almost never a helpful name for things becuase it doesn't say much about what the compnent does. Consdier using a name that describes what it does.

Also, I think you can acheive the same thing with `className='animate animate-in fade-in-10'`

Reply:
Gotcha let me try it out now!